andela/odin-ah-backend
# ft  Sign up api verification email

# #160452475 Sign up api verification email 

#### What does this PR do?
Adds signup API controller, middleware and email verification function

#### Description of Task to be completed?
- Get sendGrid token
- Adds signup controller
- Adds signup route
- Adds signup validation middleware
- Adds test for signup controller and middleware
- Adds  function to generate one time token
- Adds a class for sending emails with sendGrid
#### How should this be manually tested?
- run `npm install` in the terminal  and run `localhost:3000/api/v1/auth/signup` on Postman
- check email inbox for verification link and copy the token e.g `nwmr7d3e7zgnwmr7d3e7zg` and then
-  run `localhost:3000/api/v1/auth/confirmation/nwmr7d3e7zgnwmr7d3e7zg`
#### Any background context you want to provide?
- When testing this feature, use Gmail  as sendGrid can't send mail to Yahoo mail
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/160452475
#### Screenshots (if appropriate)
Nil
#### Questions:
Nil
